src/operations/retakeExamParticipants/RetakeExamParticipantShow.tsx

```diff
@@ -1,12 +1,31 @@
-import {Box} from "@mui/material";
+import {Box, Breadcrumbs, Link, Typography} from "@mui/material";
 import {Show} from "react-admin";
-import {useParams} from "react-router-dom";
+import {Link as RouterLink, useLocation, useParams} from "react-router-dom";
 import {CourseInfo, RetakeExamParticipantList} from "./components";
 
 export const RetakeExamParticipantShow = () => {
-  const courseId = useParams()?.id;
+  const {id: courseId} = useParams();
+  const location = useLocation();
+  const sessionId = (location.state as {sessionId?: string})?.sessionId;
```

Comment:
(location.state is typed as `any` anyway)
```suggestion
  const sessionId = location.state?.sessionId;
```